src/Symfony/Component/DependencyInjection/Loader/Configurator/Traits/FactoryTrait.php

```diff
@@ -22,8 +22,29 @@ trait FactoryTrait
      *
      * @return $this
      */
-    final public function factory(string|array|ReferenceConfigurator|Expression $factory): static
+    final public function factory(string|array|ReferenceConfigurator|Expression|\Closure $factory): static
     {
+        if ($factory instanceof \Closure) {
+            $function = new \ReflectionFunction($factory);
+            if ($function->isAnonymous()) {
+                throw new InvalidArgumentException(\sprintf('The closure "%s" passed as factory cannot be an anonymous function.', $function->getName()));
+            }
+
+            if ($function->getClosureThis()) {
+                throw new InvalidArgumentException(\sprintf('The closure "%s" passed as factory cannot be a bound closure.', $function->getName()));
```

Comment:
It's say it has to be static

Reply:
I'll change to this:
```suggestion
                throw new InvalidArgumentException(\sprintf('The closure "%s::%s" passed as factory must be a static method or a global function.', $class->name, $function->getName()));
```